editor/store/selectors.js

```diff
@@ -351,7 +354,7 @@ export function isCurrentPostPublished( state ) {
  */
 export function isEditedPostPublishable( state ) {
 	const post = getCurrentPost( state );
-	return isEditedPostDirty( state ) || [ 'publish', 'private', 'future' ].indexOf( post.status ) === -1;
+	return isEditedPostDirty( state ) || hasMetaBoxes( state ) || [ 'publish', 'private', 'future' ].indexOf( post.status ) === -1;
```

Comment:
Might recommend splitting this long line:

```js
return (
	isEditedPostDirty( state ) ||
	hasMetaBoxes( state ) ||
	[ 'publish', 'private', 'future' ].indexOf( post.status ) === -1
);
```

Minor: I like semantic readability of `includes`:

```js
return (
	isEditedPostDirty( state ) ||
	hasMetaBoxes( state ) ||
	! includes( [ 'publish', 'private', 'future' ], post.status )
);
```